### PR TITLE
[cross_file] Fix base class nullability

### PR DESCRIPTION
**Pull request by stuartmorgan**
Without this, the dummy ("interface") XFile implementation of these properties has different nullability than the others, and the analyzer doesn't match what the runtime actually sees.

## Pre-launch Checklist

- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning

### CODE REVIEW

**Comment by stuartmorgan** [comment]
Adjusted versions to make this a breaking change, since it will probably create new analyzer warnings for packages that were already migrated (although those warnings are good, because they should match the actual runtime behavior, unlike what previous versions would see).

File: packages/cross_file/pubspec.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 name: cross_file
 description: An abstraction to allow working with files across multiple platforms.
 homepage: https://github.com/flutter/plugins/tree/master/packages/cross_file
-version: 0.3.0
+version: 0.4.0
```
Comment (ditman) [inline packages/cross_file/pubspec.yaml:4]: This requires updating the pubspec of plugins that use this package

**Review by ditman — APPROVED** [review]
LGTM!

**Comment by stuartmorgan** [comment]
Per discussion, changed back to making it a non-breaking change. Camera, which is the only null-safe thing using this so far, doesn't seem to see the same analyzer errors I did.

Technically it could be an issue for clients of camera. but since this is already semi-broken for anyone using it I'd rather just push it as a bugfix if that doesn't break our CI.

**Pull request merged**